Sponsorship

yungalyx opened the pull request:
still working on templating and having editable modals

ianmah commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,168 @@
+const HeaderCard = styled.div`
+    padding: 26px 28px;
+    font-weight: bold;
+    font-family: Apercu Pro;
```
We can probably set the font family higher up so we don't have to redefine it for every component

ianmah commented on pages/sponsorship.js:
```diff
@@ -18,17 +18,11 @@ const customStyles = {
-    background-color: #FFFFFF;
-`
-
+// TODO: move font family higher, reduce need to redefine
```
let's just remove the font for now

ianmah commented:
Table doesn't seem to be lining up properly
![image](https://user-images.githubusercontent.com/11839441/87708268-4f048580-c757-11ea-91cc-f838bc3940e0.png)

ianmah commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,170 @@
+    }
+
+    
+    handleEdit = (e) => {
+        e.preventDefault();
+        //TODO: pop-up window 
+        this.setState({showEditWindow: true})
+    };
```
A bit more modern and cleaner to use this syntax for your buttons
`<Button onClick={() => this.handleEdit()}>`

Then you can do 
```suggestion
    handleEdit = () => {
        //TODO: pop-up window 
        this.setState({showEditWindow: true})
    };
```

yungalyx commented:
@ianmah styling within the table cards is up to devs, i think theyre a bit different for each page

ianmah commented:
Shouldn't the content at least line up with it's label? I think that's needed for every table

ianmah commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,203 @@
+    }
+
+    
+    handleEdit = () => {
+        //TODO: pop-up window 
+        this.setState({showEditWindow: true})
+    };
+
+    handleNew = (e) => {
+        //TODO: pop-up window 
+        this.setState({showEditWindow: true})
+    };
+    
+
+    // allows a form with "saved state" so if users close the modal, it allows
+    handleChange = (event) => {
+        this.setState({ newobj: {
+            ... this.state.newobj,
+            [event.target.name]: event.target.value} });
+        console.log(this.state.newobj);
+       
+    }   
+
+    handleCancel = () => { 
+        this.setState({showEditWindow:false});
+    }
+
+    // sends and adds it to the list
+    handleSave = (event) => {
+        // prevents refreshing of the page
```
all of these functions can be updated to use the new syntax

yungalyx commented:
@ianmah okok ill swap it over to maps, CMS figma made it seem like we could have multiple sponsors with same names 
<img width="336" alt="figma" src="https://user-images.githubusercontent.com/52767711/87975132-10850880-ca80-11ea-94d4-433d209b42f4.png">

yungalyx commented:
I only need to fix the fonts, timestamp format, and use correct images for the action buttons
+ the table dynamically configures to space items correctly based on browser size
<img width="935" alt="yeet1" src="https://user-images.githubusercontent.com/52767711/89379293-6a094c00-d6aa-11ea-8e0d-07577d9499d7.png">


Sponsorship page updates correctly with CRUD firebase functionality in firebase.js 
<img width="726" alt="yeet2" src="https://user-images.githubusercontent.com/52767711/89379373-92914600-d6aa-11ea-8f76-191bb8873e6b.png">

Waiting on modal component from Jenny's modal branch, but the form is completely working (with show and hide methods)
<img width="158" alt="yeet3" src="https://user-images.githubusercontent.com/52767711/89379385-96bd6380-d6aa-11ea-9fcb-935b41db225a.png">

yungalyx commented:
I only need to fix the fonts, timestamp format, and use correct images for the action buttons
+ the table dynamically configures to space items correctly based on browser size
<img width="935" alt="yeet1" src="https://user-images.githubusercontent.com/52767711/89379293-6a094c00-d6aa-11ea-8e0d-07577d9499d7.png">


Sponsorship page updates correctly with CRUD firebase functionality in firebase.js 
<img width="726" alt="yeet2" src="https://user-images.githubusercontent.com/52767711/89379373-92914600-d6aa-11ea-8f76-191bb8873e6b.png">

Waiting on modal component from Jenny's modal branch, but the form is completely working (with show and hide methods)
<img width="158" alt="yeet3" src="https://user-images.githubusercontent.com/52767711/89379385-96bd6380-d6aa-11ea-9fcb-935b41db225a.png">

Ciuffi commented:
Wow, looks amazing.

I'll take a look at the code tomorrow.

for now could you merge master into your branch so the CI can run?

Ciuffi commented:
looks like you merged your local master. make sure to pull master before merging because this branch still has conflicts

Ciuffi commented on utility/firebase.js:
```diff
@@ -199,43 +199,33 @@ const fireDb = {
-      await altRef.delete();
-    }
-    return true;
+  },
+  alexgetSponsors: async(website) => { 
+    var A = {};
+    const B = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
+    B.docs.forEach(doc => {A[doc.id] = doc.data()})
+    return A;
```
A more functional way to do this would be like this:
```javascript
const refs = await db.collection("Hackathons").doc(website).collection("Sponsors").get();
const docs = refs.map(doc => doc.data())
return docs;
```

also you can just delete the old get function and rename this one to getSponsors

Ciuffi commented on utility/firebase.js:
```diff
@@ -199,43 +199,33 @@ const fireDb = {
-      return false;
-    }
-    const ref = storage.ref(`${website}/${image}`);
+  deleteSponsor: async(website, sponsorName) => {
+    const ref = db
+      .collection("Hackathons")
+      .doc(website)
+      .collection("Sponsors")
+      .doc(sponsorName);
```
This delete function won't work properly because sponsors require an image upload (we host the images). I'm not sure if we're changing this but if not then we'll have to handle deleting the images from storage as well (like the original function did).

Ciuffi commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,238 @@
+    flex: 1;
+`
+
+class SponsorshipPage extends React.Component {
```
I think we're using functional components for this project (it seems like the other PRs so far have been all functional components).

I know its a lot of work but we should stay consistent, could you switch this to a functional component?

Ciuffi commented on pages/index.js:
```diff
@@ -1,13 +1,130 @@
+`
+
+const Wrapper = styled.div`
+    width: 365px;
+    margin: auto;
+`
+
+const UserNameInput = styled.input`
+    position: absolute;
+    top: 326px;
+    width: 365px;
+    height: 40px;
+    background: ${ COLOR.WHITE };
+    border: 1px solid #606060;
+    box-sizing: border-box;
+    border-radius: 2px;
+    font-size: 16px;
+    line-height: 20px;
+`
+
+const PasswordInput = styled.input`
+    position: absolute;
+    top: 441px;
+    width: 365px;
+    height: 40px;
+    background: ${ COLOR.WHITE };
+    border: 1px solid #606060;
+    box-sizing: border-box;
+    border-radius: 2px;
+    font-size: 16px;
+    line-height: 20px;
+`
+
 
 export default function Home() {
+
+  const googleSignIn = async (e) => {
+    e.preventDefault();
+    if (!firebase.apps.length) {
+      console.log(process.env.NEXT_PUBLIC_FIREBASE_API_KEY)
```
Home page should be on a different branch, not sure why its in this PR

Ciuffi commented on components/modal.js:
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+export default class Modal extends React.Component {
+  render() {
+      if(!this.props.show) {
+          return null;
+      }
+    return <div>
+        <div>{this.props.children}</div>
+
+        </div>
+
+  }
+}
```
Again, ideally this would be a functional component.

Ciuffi commented on components/card.js:
```diff
@@ -27,4 +27,23 @@ export const CardContent = styled.div`
 `;
 
-export default ({ children }) => <Card>{children}</Card>;
+export const CardDiv = styled.div`
+    display: flex;
+    flex-flow: row nowrap;
+    justify-content: space-around;
+    padding: 26px 28px;
+    border-bottom: 2px solid #EDEDED;
+`
+export const TableDiv = styled.div`
+    border: 1px solid #000000;
+    border-radius: 3px;
+    padding: 1px 1px;
+    background-color: #FFFFFF;
+`
+
+
+export default ({children}) => (
+    <Card>
+        {children}
+    </Card>
+)
```
are these card changes compatible with everyone else's use of the card component? 

Also card component changes should be in another PR

Ciuffi commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,238 @@
+                    </CardContent>
+                </Card> 
+        
+                <form onSubmit={this.handleSave}>
```
is the hide/show for this form finnished yet?

yungalyx replied:
not yet! almost done

yungalyx commented:
UPDATE SEP 5, 2020: 

1.  Corrected styling for table
<img width="946" alt="sponsooo" src="https://user-images.githubusercontent.com/52767711/92316496-bd243680-efa9-11ea-9077-902f622a865b.png">

2. changed Class component into functional component
- yeah this one kinda hurt 
<img width="324" alt="exportdefauly" src="https://user-images.githubusercontent.com/52767711/92316640-09707600-efac-11ea-80ae-1292d4d3ec91.png">


3. backend 
- simplified page creation just by specifying which hackathon event through `<SponsorshipPage name={EVENT} /> ` the routing in database uses the name to organize objects. 

<img width="501" alt="backend" src="https://user-images.githubusercontent.com/52767711/92316576-088b1480-efab-11ea-9573-076290243aee.png">

4. integrated + customized jenny's modal component
<img width="467" alt="dior" src="https://user-images.githubusercontent.com/52767711/92316505-d2996080-efa9-11ea-8847-f82b0e38e3e5.png">

There are some small styling issues, I had to use a custom createRef <input type='file'/>, so we will see quirks like being able to 'change' images or tiers in the VIEW modal. (but these changes don't actually carry through)


-- 

FEW MORE THINGS TO DO:
- [ ] add fonts
- [ ] fix date format
- [ ] make all input fields 'REQUIRED'.

yungalyx commented:
o i also added sponsorship TIERS

Ciuffi reviewed (changes requested):
This looks really great! Thanks for switching the component over to functional. I think it looks a lot cleaner that way.

A few things before we merge: Please move the sponsorship page into the [id] folder and set it up to work properly with the Page component. (You can properly just put everything inside sponsorship.js into [id]/sponsors.js). Make sure to grab the hackathon id from the props (you'll see this in sponsors.js and it'll look like this)
```javascript
export default ({ id, hackathons }) => (
  <Page currentPath={id} hackathons={hackathons}>
    Sponsors
  </Page>
);
```

Also, as we discussed at our last meeting, please also rename the sponsor.js file in the [id] folder to spocos.js so adblockers don't stop it from running lol.

Otherwise great work, seems like solid code and it looks great too. Oh, and make sure to fix the linting errors as well.

panjenny0 commented on components/card.js:
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
 `;
 
-export default ({ children }) => <Card>{children}</Card>;
+// padding: top, right, bottom, left
+// jsutify contente space around
```
```suggestion
```

panjenny0 commented on components/card.js:
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
+    justify-content: space-around;
+    align-items: center;
+    padding: ${props => props.padding};
+    border-bottom: 2px solid #EDEDED;
```
```suggestion
    border-bottom: 2px solid ${COLOR.GRAY};
```

panjenny0 commented on components/card.js:
```diff
@@ -27,4 +27,26 @@ export const CardContent = styled.div`
+    border-bottom: 2px solid #EDEDED;
+`
+export const TableDiv = styled.div`
+    border: 1px solid #000000;
```
```suggestion
    border: 1px solid ${COLOR.BLACK};
```

panjenny0 commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,301 @@
+    const loadFirebase = async() => {
+        const data = await fireDb.getSponsors(props.name);
+        setSponsors(data)
+        console.log(sponsors)
```
```suggestion
```

panjenny0 commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,301 @@
+    
+    
+    const handleEdit = (id) => {
+        //TODO: pop-up window 
```
Is this TODO still needed?
```suggestion
```

panjenny0 commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,301 @@
+            "lastmod": sponsors[id].lastmod, 
+            "tier": sponsors[id].tier,
+        });
+        // TODO
```
```suggestion
```

panjenny0 commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,301 @@
+
+    const [sponsors, setSponsors] = useState({});
+
+    const [newobj, setnewobj] = useState({
+        name: "",
+        link: "",
+        imgURL: "", 
+        lastmod: "",
+        tier: ""
+    })
+    const [imgObject, setimgObject] = useState({});
+    const [imgFile, setimgFile] = useState({});
+    const [showEditWindow, setShowEditWindow] = useState(false);
+    const [modalAction, setmodalAction] = useState({});
```
Nit: camelCase for consistency
```suggestion
    const [newobj, setNewObj] = useState({
        name: "",
        link: "",
        imgURL: "", 
        lastMod: "",
        tier: ""
    })
    const [imgObject, setImgObject] = useState({});
    const [imgFile, setImgFile] = useState({});
    const [showEditWindow, setShowEditWindow] = useState(false);
    const [modalAction, setModalAction] = useState({});
```

panjenny0 commented on pages/sponsorship.js:
```diff
@@ -0,0 +1,301 @@
+            [`${newobj.name}`]: newobj
+        }));
+
+        /*
+        this.state.sponsors[this.state.newobj.name] = this.state.newobj;
+        this.setState({ sponsors: this.state.sponsors });
+        */
+
```
```suggestion
```

panjenny0 reviewed (changes requested):
Just some very minor things but otherwise looks good!
